### PR TITLE
bugfix and logical improvement in RoomConfigDialog

### PR DESCRIPTION
**Pull request by shinespeciall**

### CODE REVIEW

File: Dialog/RoomPreviewGraphicsView.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ class RoomPreviewGraphicsView : public QGraphicsView
 
 private:
     bool initialized = false; // Used for initializing the scrollbars
-    void *dataPointers[3]; // Tilset, BG, L0
+    void *dataPointers[4] = {0, 0, 0, 0}; // Tilset, BG, L0
```
Comment (Goldensunboy) [inline Dialog/RoomPreviewGraphicsView.h]: Why are there 4? Also, this is already initialized at RoomPreviewGraphicsView.cpp:19

File: Dialog/RoomConfigDialog.cpp [hunk 2]
```diff
@@ -218,12 +252,24 @@ void RoomConfigDialog::on_ComboBox_TilesetID_currentIndexChanged(int index)
...
         {
-            elements << QString::number(WL4Constants::BGLayerDisableDefaultPtr, 16).toUpper();
+            elements << QString::number(WL4Constants::BGLayerDefaultPtr, 16).toUpper();
         }
         ui->ComboBox_BGLayerPicker->addItems(elements);
 
+        ui->CheckBox_BGLayerEnable->setChecked(true);
+        ui->CheckBox_Layer0Enable->setChecked(false); ui->CheckBox_Layer0Enable->setChecked(true);
+        ui->CheckBox_Layer2Enable->setChecked(true);
+        if(ui->ComboBox_TilesetID->currentIndex() == 0x21)
```
Comment (Goldensunboy) [inline Dialog/RoomConfigDialog.cpp]: Please add a comment describing the significance of "0x21" or make it a constant in WL4Constants

Comment (Goldensunboy) [reply Dialog/RoomConfigDialog.cpp]: 0x21 is also used on line 181

File: Dialog/RoomConfigDialog.cpp [hunk 3]
```diff
@@ -71,7 +71,9 @@ RoomConfigDialog::RoomConfigDialog(QWidget *parent, DialogParams::RoomConfigPara
     ui->graphicsView->infoLabel = ui->graphicViewDetailsLabel;
     int _tilesetPtr = WL4Constants::TilesetDataTable + CurrentRoomParams->CurrentTilesetIndex * 36;
     currentTileset = new LevelComponents::Tileset(_tilesetPtr, CurrentRoomParams->CurrentTilesetIndex);
-    ui->graphicsView->UpdateGraphicsItems(currentTileset, CurrentRoomParams->BackgroundLayerDataPtr, CurrentRoomParams->Layer0DataPtr);
+    int L0ptr = 0;
+    if(ui->ComboBox_Layer0MappingType->currentIndex() == 1) L0ptr = CurrentRoomParams->Layer0DataPtr;
```
Comment (Goldensunboy) [inline Dialog/RoomConfigDialog.cpp]: L0ptr = ui->ComboBox_Layer0MappingType->currentIndex() ? CurrentRoomParams->Layer0DataPtr : 0;

File: Dialog/RoomConfigDialog.cpp (hunk 2 above)
Comment (shinespeciall) [reply Dialog/RoomConfigDialog.cpp]: done

File: Dialog/RoomConfigDialog.cpp (hunk 3 above)
Comment (shinespeciall) [reply Dialog/RoomConfigDialog.cpp]: done

**Pull request merged**